aips/aip-xx.md

```diff
@@ -0,0 +1,198 @@
+
+## Testing
+
+Given the broad nature of this change, it is necessary to conduct extensive testing across all layers of the software
+stack. This includes: (i) individual unit tests for new components; (ii) integration tests for the consensus publisher
+and observer; (iii) end-to-end tests for network synchronization (e.g., via the existing smoke test infrastructure);
+(iv) performance tests to measure and verify the latency improvements of CO (e.g., via the existing performance test
+infrastructure in `forge`); and finally (v) pre-production tests in our `devnet` and `testnet` environments, before
+`mainnet` deployment.
+
+It is also particularly important to ensure that sufficient testing takes place for failure scenarios and edge cases.
+For example, node restarts, node failures, data wipes, invalid or malicious network messages, failures during 
+block execution and proof processing, etc. This will help to ensure CO is reliable enough to withstand real-world
+conditions.
+
+### Preliminary Performance Results
+
+Preliminary testing has already been conducted in several test environments, including `forge`. In these experiments,
+the performance of CO was measured and compared against state sync. The results of the experiments showed significant
+improvements in end-to-end transaction latencies for VFNs and PFNs. For example:
+1. In a semi-realistic, high-load scenario, CO reduced end-to-end transaction latencies for VFNs by `~30%`
+   (from `~2.1s` to `~1.5s`) at `~10k TPS`.
+2. In a medium-load scenario with extreme network chaos, CO reduced end-to-end transaction latencies for PFNs by
+   `~50%` (from `~3.1s` to `~1.6s`) at `~5k TPS`.
+
+## Risks and Drawbacks
+
+There are several risks and drawbacks associated with CO, including:
+1. **Transaction Re-execution**: One of the benefits of state sync is that it provides an
+   [intelligent syncing](https://aptos.dev/en/network/nodes/configure/state-sync#intelligent-syncing)
+   mode, where fullnodes can skip transaction re-execution and reuse the execution results from validators.
+   This helps to reduce CPU usage and improve synchronization time. However, CO does not
+   provide this as it requires fullnodes to re-execute blocks in real-time. As a result, it requires all fullnodes
+   to have equivalent (or better) hardware resources than validators, to ensure they can process blocks at the
+   same rates while dealing with other overheads (e.g., REST API requests).
+      - **Mitigation**: While this is a drawback, it does not pose a significant risk, as fullnodes are already
+      required to deploy equivalent hardware resources as validators. Moreover, the benefits of CO in terms of
+      reduced latencies are more than sufficient to outweigh the additional costs of transaction re-execution.
+2. **CO Complexity**: CO will introduce additional complexity to the Aptos node software, as new components
```

Comment:
Is there any extra bandwidth cost because of the new network messages?

Reply:
No, if anything it'll be less data because nodes will be syncing transactions instead of outputs. 😄